app/helpers/pagelets_helper.rb

```diff
@@ -67,6 +67,64 @@ def th_content(col)
     sort col[:key], as: col[:label], default: col[:default_sort] || 'ASC'
   end
 
+  def defined_columns
+    columns = Pagelets::Manager.pagelets_at('hosts/_list', :hosts_table_column_header).map { |pt| { key: pt.opts[:key], label: pt.opts[:label], profiles: pt.profiles.map { |pr| pr.label } } if pt.opts[:key] }.compact
+    columns.each do |column|
+      column[:checked] = @selected_columns&.include?(column[:key].to_s)
+    end
+  end
```

Comment:
```suggestion
  def defined_columns
    Pagelets::Manager.pagelets_at('hosts/_list', :hosts_table_column_header).map do |pt|
      next unless pt.opts[:key]
      {
        key: pt.opts[:key],
        label: pt.opts[:label],
        profiles: pt.profiles.map { |pr| pr.label },
        checked: @selected_columns&.include?(column[:key].to_s),
      }
    end.compact
  end
```

We could re-write it a bit for better readability.